lib/pundit.rb

```diff
@@ -88,7 +88,8 @@ def policy_scope(user, scope)
     # @raise [NotDefinedError] if the policy scope cannot be found
     # @return [Scope{#resolve}] instance of scope class which can resolve to a scope
     def policy_scope!(user, scope)
-      PolicyFinder.new(scope).scope!.new(user, scope).resolve
+      model = scope.is_a?(Array) ? scope.last : scope
```

Comment:
Interesting... It's been so long since I used Rails 3 I had not considered this.